Core/src/org/sleuthkit/autopsy/casemodule/CaseOpenAction.java

```diff
@@ -84,18 +84,29 @@ void openCaseSelectionWindow() {
         fileChooser.setFileSelectionMode(JFileChooser.FILES_ONLY);
         fileChooser.setMultiSelectionEnabled(false);
         fileChooser.setFileFilter(caseMetadataFileFilter);
+
         if (null != ModuleSettings.getConfigSetting(ModuleSettings.MAIN_SETTINGS, PROP_BASECASE)) {
             fileChooser.setCurrentDirectory(new File(ModuleSettings.getConfigSetting("Case", PROP_BASECASE))); //NON-NLS
         }
-        
+
+        /**
+         * If the open multi user case dialog is open make sure it's not set
+         * to always be on top as this hides the file chooser on macOS.
+         */
+        OpenMultiUserCaseDialog.getInstance().setAlwaysOnTop(false);
         String optionsDlgTitle = NbBundle.getMessage(Case.class, "CloseCaseWhileIngesting.Warning.title");
         String optionsDlgMessage = NbBundle.getMessage(Case.class, "CloseCaseWhileIngesting.Warning");
         if (IngestRunningCheck.checkAndConfirmProceed(optionsDlgTitle, optionsDlgMessage)) {
             /**
              * Pop up a file chooser to allow the user to select a case metadata
              * file (.aut file).
              */
-            int retval = fileChooser.showOpenDialog(WindowManager.getDefault().getMainWindow());
+            /**
+             * Passing the fileChooser as its own parent gets around an issue
+             * where the fileChooser was hidden behind the CueBannerPanel ("Welcome" dialog)
+             * on macOS.
+             */
+            int retval = fileChooser.showOpenDialog(fileChooser);
```

Comment:
This change causes us to re-introduce the issue where the dialog will be opened on the main display instead of over the current location of Autopsy. This does not strike me as a deal breaker but as it is a regression of sorts I wanted to make sure it was noted.

Reply:
I believe I've addressed this so it's ready for review again.